types/list.go

```diff
@@ -15,10 +15,28 @@ import (
 )
 
 var (
-	_ attr.Type  = ListType{}
-	_ attr.Value = &List{}
+	_ ListTypable  = ListType{}
+	_ ListValuable = &List{}
 )
 
+// ListTypable extends attr.Type for list type types.
+// Implement this interface to create a custom ListType type type.
```

Comment:
Nit: I think we can just refer to the "type type" as just "type" and leave the distinction of "value type" to be separate 😎 

```suggestion
// ListTypable extends attr.Type for list types.
// Implement this interface to create a custom ListType type.
```